simpl/config.py

```diff
@@ -466,12 +467,107 @@ def build_parser(self, options, permissive=False, **override_kwargs):
+        :raises: SystemExit
+        """
+        options = []
+        for option in self._options:
+            kwargs = option.kwargs.copy()
+            if option.name in values:
+                if 'default' in kwargs:
+                    # Since we're overriding defaults, we need to
+                    # preserve the default value for the help text:
+                    help_text = kwargs.get('help')
+                    if help_text:
+                        if '(default: ' not in help_text:
+                            kwargs['help'] = '%s (default: %s)' % (
+                                help_text, kwargs['default']
+                            )
+                kwargs['default'] = values[option.name]
+                kwargs['required'] = False  # since we have a value
+            temp = Option(*option.args, **kwargs)
+            options.append(temp)
+        parser = self.build_parser(options,
+                                   formatter_class=argparse.HelpFormatter)
+        if argv:
+            parsed, extras = parser.parse_known_args(argv[1:])
+            if extras:
+                valid, _ = self.parse_passthru_args(argv[1:])
+                parsed, extras = parser.parse_known_args(valid)
+                if extras and strict:  # still
+                    self.build_parser(options)
+                    parser.parse_args(argv[1:])
+        else:
+            parsed = parser.parse_args([])
+
+        results = vars(parsed)
+        raise_for_group = {}
+        for option in self._options:
+            if option.kwargs.get('required'):
+                if option.dest not in results or results[option.dest] is None:
+                    if getattr(option, '_mutexgroup', None):
+                        raise_for_group.setdefault(option._mutexgroup, [])
+                        raise_for_group[option._mutexgroup].append(
+                            option._action)
+                    else:
+                        raise SystemExit("'%s' is required. See --help "
+                                         "for more info." % option.name)
+                else:
+                    if getattr(option, '_mutexgroup', None):
+                        raise_for_group.pop(option._mutexgroup, None)
+        if raise_for_group:
+            optstrings = [str(k.option_strings)
+                          for k in raise_for_group.values()[0]]
+            msg = "One of %s required. " % " ,".join(optstrings)
+            raise SystemExit(msg + "See --help for more info.")
+        return results
+
     def parse_cli(self, argv=None, permissive=False):
         """Parse command-line arguments into values.
 
         :keyword permissive: when true, does not validate required or extra
             arguments.
         """
+        warnings.warn("Calling `cli_values` directly is deprecated. Call "
+                      "`cli_values` and/or `validate_config` instead.",
+                      DeprecationWarning)
```

Comment:
I think you meant `` Calling `parse_cli` directly is deprecated. `` Maybe also mention it is scheduled for removal in 1.0.0 or something?